File: amazia-core/src/main/java/com/example/GetProduct/controller/GetProductController.java
package com.example.GetProduct.controller;

import com.example.GetProduct.service.GetProductService;
import com.example.shared.entity.Product;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api")
public class GetProductController {

    private final GetProductService getProductService;

    public GetProductController(GetProductService getProductService) {
        this.getProductService = getProductService;
    }

    @GetMapping("/products")
    public List<Product> getPublished() {
        return getProductService.getPublished();
    }

    @GetMapping("/admin/products")
    public List<Product> getAll() {
        return getProductService.getAll();
    }

    @GetMapping("/products/{id}")
    public ResponseEntity<Product> getById(@PathVariable Long id) {
        return ResponseEntity.ok(getProductService.getById(id));
    }
}

File: amazia-console/app/Product/Controller/GetProductController.php
<?php

namespace App\Product\Controller;

use App\Http\Controllers\Controller;
use App\Product\Service\GetProductService;

class GetProductController extends Controller
{
    private GetProductService $service;

    public function __construct(GetProductService $service)
    {
        $this->service = $service;
    }

    public function __invoke(int $id)
    {
        $response = $this->service->getById($id);
        return response()->json($response->json(), $response->status());
    }
}
